src/main/java/bio/terra/cli/utils/MountUtils.java

```diff
@@ -0,0 +1,241 @@
+      }
+    }
+    return resourceMountPaths;
+  }
+
+  /**
+   * Helper method to get mountable resources in the workspace, any file containing resource or
+   * reference to a file containing resource.
+   */
+  public static List<Resource> getMountableResources() {
+    return Context.requireWorkspace().listResources().stream()
+        .filter(
+            r -> {
+              if (r.getResourceType() == Resource.Type.GCS_BUCKET) {
+                return true;
+              } else if (r.getResourceType() == Resource.Type.GCS_OBJECT) {
+                try {
+                  return ((GcsObject) r).isPrefix();
+                } catch (SystemException e) {
+                  // Pass through GCS objects that are inaccessible to display error on mounted
+                  // folder later
+                  return true;
+                }
+              }
+              return false;
+            })
+        .toList();
+  }
+
+  /**
+   * Helper method to get workspace folder paths. Used to build the mount paths for mountable
+   * resources.
+   *
+   * @return A map of folder IDs to folder paths.
+   */
+  private static Map<UUID, Path> getFolderPaths() {
+    List<Folder> folders = Context.requireWorkspace().listFolders();
+    Map<UUID, Path> folderPaths = new HashMap<>();
+
+    for (Folder folder : folders) {
+      Path path = Paths.get(folder.getDisplayName());
+
+      // Recursively prepend parent folder names to path
+      UUID parentId = folder.getParentFolderId();
+      while (parentId != null) {
+        // Get parent folder
+        Folder parent = null;
+        for (Folder f : folders) {
+          if (f.getId().equals(parentId)) {
+            parent = f;
+            break;
+          }
+        }
+        // Prepend parent folder name to path
+        if (parent != null) {
+          path = Paths.get(parent.getDisplayName()).resolve(path);
+          parentId = parent.getParentFolderId();
+        }
+      }
+      folderPaths.put(folder.getId(), path);
+    }
+    return folderPaths;
+  }
+
+  /**
+   * Creates local directories for all the given resource paths.
+   *
+   * @param paths A list of resource paths to create directories for.
+   * @throws SystemException If the creation of the directories failed.
+   */
+  public static void createResourceDirectories(List<Path> paths) throws SystemException {
+    for (Path path : paths) {
+      if (!path.toFile().exists()) {
+        if (!path.toFile().mkdirs()) {
+          throw new SystemException("Failed to create directory: " + path);
+        }
+      }
+    }
+  }
+
+  /**
+   * Recursively deletes all empty subdirectories in the WORKSPACE_DIR, excluding the WORKSPACE_DIR
+   * itself.
+   *
+   * <p>Throws UserActionableException if a non-empty directory is encountered. Throws
+   * SystemException if there is an error during deletion.
+   */
+  public static void deleteResourceDirectories() {
```

Comment:
A similar implementation is in the test context: https://github.com/DataBiosphere/terra-cli/blob/main/src/test/java/harness/TestContext.java#L90. Am curious if there's a pattern to share utility functions between testing and main.

Reply:
yeah test can use this one too if you want to migrate those usage.